Source/Request/CMRequest.m

```diff
@@ -676,6 +676,13 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
 
 
 - (NSURLRequest *)connection:(NSURLConnection *)connection willSendRequest:(NSURLRequest *)request redirectResponse:(NSURLResponse *)response {
+	NSHTTPURLResponse *httpURLResponse = (NSHTTPURLResponse *)response;
+	if (httpURLResponse) {
+		if (httpURLResponse.statusCode >= kHTTPStatusMultipleChoices && httpURLResponse.statusCode < kHTTPStatusBadRequest) {
+			// it's a redirect
+			self.redirectURL = [[httpURLResponse allHeaderFields] valueForKey:@"Location"];
+		}
```

Comment:
Add kCumulusHTTPHeaderLocation to the list of header constants in CMConstants.h and use that instead.